storage/storage_dest.go

```diff
@@ -920,12 +919,27 @@ func (s *storageImageDestination) createNewLayer(index int, layerDigest digest.D
 				return nil, nil
 			}
 
+			// FIXME: Should we insist on UncompressedDigest being always set, and hard fail otherwise??
 			untrustedUncompressedDigest = d
 			// While the contents of the digest are untrusted, make sure at least the _format_ is valid,
 			// because we are going to write it to durable storage in expectedLayerDiffIDFlag .
 			if err := untrustedUncompressedDigest.Validate(); err != nil {
 				return nil, err
 			}
+		} else {
+			// FIXME: Clean up. Maybe the generic code can provide us the config earlier?
+			// FIXME: Always enforce this for all layers??!
+			d, err := s.untrustedLayerDiffID(index)
+			if err != nil {
+				return nil, err
```

Comment:
All of this completely breaks for non-OCI images: `determining DiffID for manifest type %q is not yet supported`

This needs to be implemented for v2s2, and we need a reasonable fallback for v2s1.